cmd/edit/machinepool/nodepool.go

```diff
@@ -307,13 +307,26 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
 	asBuilder := cmv1.NewNodePoolAutoscaling()
 	changed := false
 
-	if nodePool.Autoscaling().MinReplica() != minReplicas && minReplicas >= 1 {
-		asBuilder = asBuilder.MinReplica(minReplicas)
-		changed = true
+	existingMinReplica := nodePool.Autoscaling().MinReplica()
+	existingMaxReplica := nodePool.Autoscaling().MaxReplica()
+
+	if existingMinReplica != minReplicas {
+		if minReplicas >= 1 {
+			asBuilder = asBuilder.MinReplica(minReplicas)
+			changed = true
+		}
+		if minReplicas == 0 {
+			asBuilder = asBuilder.MinReplica(existingMinReplica)
+		}
 	}
-	if nodePool.Autoscaling().MaxReplica() != maxReplicas && maxReplicas >= 1 {
-		asBuilder = asBuilder.MaxReplica(maxReplicas)
-		changed = true
+	if existingMaxReplica != maxReplicas {
+		if maxReplicas >= 1 {
+			asBuilder = asBuilder.MaxReplica(maxReplicas)
+			changed = true
+		}
+		if maxReplicas == 0 {
+			asBuilder = asBuilder.MaxReplica(existingMaxReplica)
```

Comment:
Would it be ok to set it to the existing values initially instead to save on some condition branching?

Reply:
Haha I actually did that first and refactored it to what it looks like now - will make the change